### PR TITLE
add-node using IPv6 address

### LINKED ISSUE #1751: Cannot join over ipv6
It seems the `microk8s join` script doesn't handle ipv6 addresses at all. My nodes use an ipv6 overlay network to communicate and cannot reach each other over ipv4. Maybe adding a -6 flag is a good solution, or potentially detecting the ipv6 format.

```
ubuntu@mm-cluster-stephen-do:~$ microk8s join [201:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx:xxxx]:25000/xxxxxxxxxxx
Contacting cluster at [201
Traceback (most recent call last):
  File "/snap/microk8s/1791/scripts/cluster/join.py", line 967, in <module>
    join_dqlite(connection_parts)
  File "/snap/microk8s/1791/scripts/cluster/join.py", line 871, in join_dqlite
    info = get_connection_info(master_ip, master_port, token, cluster_type="dqlite")
  File "/snap/microk8s/1791/scripts/cluster/join.py", line 70, in get_connection_info
    verify=False,
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/requests/api.py", line 107, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/requests/api.py", line 53, in request
    return session.request(method=method, url=url, **kwargs)
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/requests/sessions.py", line 466, in request
    prep = self.prepare_request(req)
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/requests/sessions.py", line 400, in prepare_request
    hooks=merge_hooks(request.hooks, self.hooks),
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/requests/models.py", line 293, in prepare
    self.prepare_url(url, params)
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/requests/models.py", line 345, in prepare_url
    scheme, auth, host, port, path, query, fragment = parse_url(url)
  File "/snap/microk8s/1791/usr/lib/python3/dist-packages/urllib3/util/url.py", line 184, in parse_url
    host, url = url.split(']', 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

I tried to fix it with this patch:
```
diff --git a/scripts/cluster/join.py b/scripts/cluster/join.py
index 9ba826d..269bc9a 100755
--- a/scripts/cluster/join.py
+++ b/scripts/cluster/join.py
@@ -64,8 +64,11 @@ def get_connection_info(master_ip, master_port, token, callback_token=None, clus
 
         # TODO: enable ssl verification
         try:
+            format_string = "https://{}:{}/{}/join"
+            if master_ip.find(":"):
+                format_string = "https://[{}]:{}/{}/join"
             connection_info = requests.post(
-                "https://{}:{}/{}/join".format(master_ip, master_port, CLUSTER_API_V2),
+                format_string.format(master_ip, master_port, CLUSTER_API_V2),
                 json=req_data,
                 verify=False,
             )  # type: requests.models.Response
@@ -83,8 +86,11 @@ def get_connection_info(master_ip, master_port, token, callback_token=None, clus
 
         # TODO: enable ssl verification
         try:
+            format_string = "https://{}:{}/{}/join"
+            if master_ip.find(":"):
+                format_string = "https://[{}]:{}/{}/join"
             connection_info = requests.post(
-                "https://{}:{}/{}/join".format(master_ip, master_port, CLUSTER_API),
+                format_string.format(master_ip, master_port, CLUSTER_API),
                 json=req_data,
                 verify=False,
             )
@@ -864,8 +870,8 @@ def join_dqlite(connection_parts):
     """
     token = connection_parts[1]
     master_ep = connection_parts[0].split(":")
-    master_ip = master_ep[0]
-    master_port = master_ep[1]
+    master_ip = ":".join(master_ep[:-1])
+    master_port = master_ep[-1]
 
     print("Contacting cluster at {}".format(master_ip))
     info = get_connection_info(master_ip, master_port, token, cluster_type="dqlite")
```

But I just realized that it seems whatever is on port 25000 is only on the lan ipv4 address of my first node. Going off [this documentation](https://microk8s.io/docs/ports) and port scanning both the v4 and v6 address, it seems that out of the box, port 25000 is the only one that is v4 only for some reason. Maybe it will work once that's fixed.

### PR DESCRIPTION
**Pull request by sachinkumarsingh092**
### Summary
Now that microk8s supports dual-stack, it is also possible to add-nodes using IPv6 addresses,

Fixes #1751

### CODE REVIEW

File: build-scripts/components/cluster-agent/version.sh [hunk 1]
```diff
@@ -1,3 +1,3 @@
 #!/bin/bash
 
-echo "main"
+echo "MK-1275/fix-ipv6-add-node"
```
Comment (sachinkumarsingh092) [inline build-scripts/components/cluster-agent/version.sh:3]: Change to main before merging.

**Comment by ktsakalozos** [comment]
@sachinkumarsingh092 could you please rebase this code? There was a bugfix on the build process of k8s causing the tests to fail.

Could you also fix the lint errors? Thanks

File: scripts/wrappers/join.py [hunk 2]
```diff
@@ -636,7 +636,11 @@ def update_dqlite(cluster_cert, cluster_key, voters, host):
     with open("{}/info.yaml".format(cluster_backup_dir)) as f:
         data = yaml.safe_load(f)
     if "Address" in data:
-        port = data["Address"].split(":")[1]
+        port = data["Address"].rsplit(":")[-1]
+
+    # If host is an IPv6 address, wrap it in square brackets
+    if host.count(":") > 0:
```
[the remark this answers is not in the document]
Comment (sachinkumarsingh092) [reply scripts/wrappers/join.py]: Done

File: scripts/wrappers/join.py [hunk 3]
```diff
@@ -636,7 +637,15 @@ def update_dqlite(cluster_cert, cluster_key, voters, host):
     with open("{}/info.yaml".format(cluster_backup_dir)) as f:
         data = yaml.safe_load(f)
     if "Address" in data:
-        port = data["Address"].split(":")[1]
+        port = data["Address"].rsplit(":")[-1]
+
+    # If host is an IPv6 address, wrap it in square brackets
+    try:
+        ip = ipaddress.ip_address(host)
+        if ip.version == 6:
```
Comment (neoaggelos) [inline scripts/wrappers/join.py:645]:
```suggestion
        if ipaddress.ip_address(host).version == 6:
```

**Review by neoaggelos — COMMENTED** [review]
Looks good, minor comment

File: build-scripts/components/cluster-agent/version.sh (hunk 1 above)
Comment (neoaggelos) [reply build-scripts/components/cluster-agent/version.sh]: PR is merged, please update

**Pull request merged**